### PR TITLE
Anchor with zero fee htlc support (aka. experimental-anchors)

### PR DESCRIPTION
**Pull request by rustyrussell**
Cleanups, new JSON fields for anchor support, anchor spending support so we can lowball our commitment tx fees, and min-emergency-msat for reserving some sats for closing anchor channels (respecting that meant quite a neatning/rework of our fundchannel and withdraw plugin ops).

I expect this to be experimental for one release, and the default in future releases.

### CODE REVIEW

File: doc/schemas/fundpsbt.request.json [hunk 1]
```diff
@@ -32,6 +32,14 @@
...
     "excess_as_change": {
       "type": "boolean"
+    },
+    "nonwrapped": {
+      "added": "v23.02",
+      "type": "boolean"
+    },
+    "keep_emergency_funds": {
+      "added": "v23.08",
+      "type": "boolean"
```
Comment (cdecker) [inline doc/schemas/fundpsbt.request.json:42]: Should this really be a boolean? This kind of assumes that all funds kept back for emergencies are always the same, while think any other HTLC construction (or PTLC) will have a different weight, and different timing requirements. Also in future we might want to keep some funds aside for each channel, or just a single emergency amount that is threaded through multiple channels, and a bool cannot express that. An `amount_msat` to hold back overall is likely more flexible.

Also naming, it's hard :-)

**Review by cdecker — APPROVED** [review]
Very nice PR, am I correct in assuming that this subsumes #6137 which can then also be closed?

This is good as is, just some minor comments about the usability, but otherwise good to go #5850 

ACK 8df044a2d06aab7a629ac7cda523d4b8e3b7ee8e

File: wallet/wallet.c [hunk 2]
```diff
@@ -604,6 +604,69 @@ struct utxo *wallet_find_utxo(const tal_t *ctx, struct wallet *w,
...
+					", scriptpubkey "
+					", reserved_til"
+					", csv_lock"
+					", is_in_coinbase"
+					" FROM outputs"
+					" WHERE status = ?"
+					" OR (status = ? AND reserved_til <= ?)"));
+	db_bind_int(stmt, 0, output_status_in_db(OUTPUT_STATE_AVAILABLE));
+	db_bind_int(stmt, 1, output_status_in_db(OUTPUT_STATE_RESERVED));
+	db_bind_u64(stmt, 2, current_blockheight);
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply wallet/wallet.c:638]: Conservative assumption: reservations are assumed to be in progress for *some* tx.

File: doc/schemas/fundpsbt.request.json (hunk 1 above)
Comment (rustyrussell) [reply doc/schemas/fundpsbt.request.json:42]: Yeah, but I think the internal logic really needs to handle those cases, though. This is simply a hook for "fundchannel" to say "I'm opening an anchor channel this this, so you need the reserve even if you don't think you do".

If we need a more sophisticated emergency reserve, that really needs to be handled elsewhere...

Comment (rustyrussell) [reply doc/schemas/fundpsbt.request.json:42]: OK, I've renamed this arg to "creating_anchor_channel", which is *explicitly* what it means, at least for now.

**Comment by rustyrussell** [comment]
master rebase, and parameter renaming inspired by @cdecker

File: lightningd/anchorspend.c [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	struct anchor_details *adet = tal(ctx, struct anchor_details);
+
+	/* If we don't have an anchor, we can't do anything. */
+	if (!channel_has(channel, OPT_ANCHOR_OUTPUTS))
+		return tal_free(adet);
+
+	if (!hsm_capable(ld, WIRE_HSMD_SIGN_ANCHORSPEND)) {
+		log_broken(ld->log, "hsm not capable of signing anchorspends!");
+		return tal_free(adet);
+	}
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply lightningd/anchorspend.c:115]: Hmm, yes, I'll check it earlier.  You don't need a new HSM if you don't use experimental-anchors, so I didn't make it required.

**Comment by rustyrussell** [comment]
Trivial rebase on master (which has CI fixes)

**Comment by rustyrussell** [comment]
Rebased on top of #6356 for CI fixes...

**Comment by rustyrussell** [comment]
Rebased on master again, with proper rebuild of the generated python files. This time for sure!

**Comment by rustyrussell** [comment]
... elements test fixes (mainly, disabling some anchor + elements tests for now: I don't want to recalc all the tx sizes!).

**Comment by rustyrussell** [comment]
Test flake fix for anchors test.

**Pull request merged**